src/main/java/edu/byu/cs/canvas/CanvasIntegration.java

```diff
@@ -151,19 +153,78 @@ public static void submitGrade(int userId, int assignmentNum, float grade, Strin
     }
 
     /**
-     * Gets the git repository url for the given user from their GitHub Repository assignment submission on canvas
+     * Submits the given grade for the given assignment for the given user
      *
-     * @param userId The canvas user id of the user to get the git repository url for
-     * @return The git repository url for the given user
+     * @requires The maps passed in must support the putIfAbsent method (Map.of() does not)
+     *
+     * @param userId            The canvas user id of the user to submit the grade for
+     * @param assignmentNum     The assignment number to submit the grade for
+     * @param grades            A Map of rubric item id's to grades for that rubric item
+     * @param rubricComments    A Map of rubric item id's to comments to put on that rubric item
+     * @param assignmentComment A comment for the entire assignment, if necessary
      * @throws CanvasException If there is an error with Canvas
      */
-    public static String getGitRepo(int userId) throws CanvasException {
-        CanvasSubmission submission = makeCanvasRequest(
+    public static void submitGrade(int userId, int assignmentNum, Map<String, Float> grades, Map<String, String> rubricComments, String assignmentComment) throws CanvasException {
+        CanvasSubmission submission = getSubmission(userId, assignmentNum);
+        if(submission.rubric_assessment() != null) {
+            for(Map.Entry<String, CanvasIntegration.RubricItem> entry : submission.rubric_assessment().items().entrySet()) {
```

Comment:
This code is smelly to me.

This control block is mutating the `grades` and `rubricComments` data structures that were originally passed in from the user based on data extracted from other portion of the app. This is scary because the callers of this function may not understand or expect that behavior (a short time after the original authorship of the code). 

If it's strictly necessary to merge in comments from these two data sources, then that should be explicitly stated and explained. However, it's likely that there is another way of accepting the comments that does one of the following:
1. Requires the user to acknowledge & control the merging process by explicitly calling a function or setting a flag
2. Collects the data from various stages of calculation without a required merging process (why are there multiple places to look anyways)?